vip-rest-api.php

```diff
@@ -68,7 +68,18 @@ function wpcom_vip_verify_go_rest_api_request_authorization( $namespace, $auth_h
  * @param  string $namespace RESET API route's namespace
  * @return bool
  */
-function wpcom_vip_go_rest_api_request_allowed( $namespace ) {
+function wpcom_vip_go_rest_api_request_allowed( $namespace, $cap = 'do_not_allow' ) {
+	// First check basic auth
+	$basic_auth_user = wpcom_vip_basic_auth_user();
+	if ( $basic_auth_user && ! is_wp_error( $basic_auth_user ) &&
+		$basic_auth_user->ID && $basic_auth_user->ID > 0 ) {
+			$user_id = $basic_auth_user->ID;
+
+			if ( user_can( $user_id, 'vip_support' ) || user_can( $user_id, $cap ) ) {
```

Comment:
> Does user_can( $user_id, 'do_not_allow' ) always return false?

Yep, `do_not_allow` is what core uses internally as well.

https://core.trac.wordpress.org/browser/tags/4.9.8/src/wp-includes/capabilities.php#L37

Reply:
It's not possible to actually have `do_not_allow` because `WP_User::has_cap` removes it before checking caps. So even if you did something like `$user->add_cap( 'do_not_allow' )`, you couldn't get through check.

https://core.trac.wordpress.org/browser/tags/4.9.8/src/wp-includes/class-wp-user.php#L747